### ISSUE TITLE
invoke cargo check with rust-project.json

### ISSUE THREAD

**Issue by nishanthkarthik** [rust-lang/rust-analyzer#16246]
When I use a bazel generated rust-project.json, `cargo check` is not invoked by rust-analyzer on save. 

When I remove `rust-project.json` and stick to a handwritten `Cargo.toml`, `cargo check` is invoked on save as expected.

Is there a way I can make `cargo check` work with `rust-project.json`? I understand the main point of `rust-project.json` is to avoid depending on `Cargo.toml` but I miss having the error-on-save development workflow.

Thanks!

**Comment by flodiebold** [comment]
IIRC if you set `rust-analyzer.check.overrideCommand` we run that even for rust-project.json, or am I misremembering?

**Comment by nishanthkarthik** [comment]
Thanks for the tip! I tried setting the `overrideCommand` using the one from the manual but this did not trigger `cargo check` on save. I use the text editor Kate with the built-in LSP client.

```json
"initializationOptions": {
        "cachePriming": { "enable": true },
        "check": { "enable": true, "allTargets": false, "overrideCommand": "cargo check --workspace --message-format=json" },
        "checkOnSave": { "enable": true },
        "diagnostics": { "enable": true, "experimental": { "enable": true } }
      }
```

With cargo.toml, my LSP logs print

```
[12:07:15 ￼ LSP Client Log] Started server rust@/ssd/pa-dhcp/rs: /usr/bin/rust-analyzer
[12:07:16 ￼ LSP Server Info] Fetching [100%]
[12:07:16 ￼ LSP Server Info] Roots Scanned [100%] 27/27
[12:07:16 ￼ LSP Server Info] Building [100%]
[12:07:16 ￼ LSP Server Info] Loading [100%]
[12:07:21 ￼ LSP Server Info] Indexing [100%]
[12:07:16 ￼ LSP Server Info] cargo check [100%]
```

With `rust-project.json`, I don't see the `cargo check` line anymore.

```
[12:11:52 ￼ LSP Client Log] Started server rust@/ssd/pa-dhcp/rs: /usr/bin/rust-analyzer
[12:11:52 ￼ LSP Server Info] Fetching [100%]
[12:11:52 ￼ LSP Server Info] Roots Scanned [100%] 26/26
[12:11:52 ￼ LSP Server Info] Building [100%]
[12:11:52 ￼ LSP Server Info] Loading [100%]
[12:11:57 ￼ LSP Server Info] Indexing [100%]
```
